packages/slice-machine/components/Forms/RenameSliceModal/RenameSliceModal.tsx

```diff
@@ -1,50 +1,65 @@
 import { Box } from "theme-ui";
-import useSliceMachineActions from "@src/modules/useSliceMachineActions";
-import ModalFormCard from "../../ModalFormCard";
-import { InputBox } from "../components/InputBox";
 import { useSelector } from "react-redux";
+
+import useSliceMachineActions from "@src/modules/useSliceMachineActions";
 import { SliceMachineStoreType } from "@src/redux/type";
-import { isModalOpen } from "@src/modules/modal";
-import { ModalKeysEnum } from "@src/modules/modal/types";
 import { getLibraries, getRemoteSlices } from "@src/modules/slices";
+import { ComponentUI } from "@lib/models/common/ComponentUI";
+import { renameSlice } from "@src/features/slices/actions/renameSlice";
+
+import { InputBox } from "../components/InputBox";
+import ModalFormCard from "../../ModalFormCard";
 import { SliceModalValues } from "../formsTypes";
 import { validateSliceModalValues } from "../formsValidator";
 
 interface RenameSliceModalProps {
-  sliceName: string;
-  sliceId: string;
-  libName: string;
+  isOpen: boolean;
+  slice?: ComponentUI;
+  onClose: () => void;
 }
 
 export const RenameSliceModal: React.FC<RenameSliceModalProps> = ({
-  sliceName,
-  sliceId,
-  libName,
+  slice,
+  isOpen,
+  onClose,
 }) => {
-  const { renameSlice, closeModals } = useSliceMachineActions();
-  const { isRenameSliceModalOpen, localLibs, remoteLibs } = useSelector(
+  const { renameSliceSuccess } = useSliceMachineActions();
+  const { localLibs, remoteLibs } = useSelector(
     (store: SliceMachineStoreType) => ({
-      isRenameSliceModalOpen: isModalOpen(store, ModalKeysEnum.RENAME_SLICE),
       localLibs: getLibraries(store),
       remoteLibs: getRemoteSlices(store),
     }),
   );
 
-  const handleOnSubmit = (values: SliceModalValues) => {
-    renameSlice(libName, sliceId, values.sliceName);
+  const handleOnSubmit = async (values: SliceModalValues) => {
+    if (slice) {
+      await renameSlice({
+        slice,
+        newSliceName: values.sliceName,
+        onSuccess: (renamedSlice) => {
+          renameSliceSuccess(
```

Comment:
Is there a reason for not passing `renamedSlice` directly? Eg. we don't want `renameSliceSuccess` to how these properties are stored?

Reply:
You're right! sliceId so `renamedSlice.model.id` is not even used as I using model.id in the reducer. So I can remove second paramater. But why I dont just pass renamedSlice is because we are working with SliceSM and not ComponentUI for the reducer and so I need the libName (first param)